### PR TITLE
DPoP release notes and documentation polishing

### LINKED ISSUE #21922: DPoP: release notes
### Description

Add release notes once all (or most of) tasks from epic #21916 are done (maybe with the exception of #21921, which is most problematic and is not a strict blocker for announce DPoP support).

### PR DESCRIPTION
**Pull request by mposolda**
closes #21922

This PR is for release notes for DPoP and also it adds some minor polishing and clarification to the DPoP documentation based on latest changes related to DPoP.

@tnorimat @andymunro Are you please able to review this?

### CODE REVIEW

**Review by tnorimat — APPROVED** [review]
@mposolda LGTM.

File: docs/documentation/server_admin/topics/clients/oidc/con-advanced-settings.adoc [hunk 1]
```diff
@@ -110,11 +110,15 @@ If this setting is on, the workflow is:
 
 If verification fails, {project_name} rejects the token.
 
-In the following cases, {project_name} will verify the client sending the refresh token:
+If the switch `OAuth 2.0 DPoP Bound Access Tokens Enabled` is off, client can still send `DPoP` proof in the token request. In that case, {project_name} will verify DPoP proof
+and will add the thumbprint to the token. But if the switch is off, DPoP binding is not enforced by {project_name} server for this client. It is recommended to have this switch
+on if you want to make sure that particular client always uses DPoP binding.
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/clients/oidc/con-advanced-settings.adoc:115]:
```suggestion
If the switch `OAuth 2.0 DPoP Bound Access Tokens Enabled` is off, the client can still send `DPoP` proof in the token request. In that case, {project_name} will verify DPoP proof
and will add the thumbprint to the token. But if the switch is off, DPoP binding is not enforced by the {project_name} server for this client. It is recommended to have this switch
on if you want to make sure that particular client always uses DPoP binding.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks just a couple of changes.

**Comment by mposolda** [comment]
@tnorimat @andymunro Thanks for the review!
@andymunro I've applied the changes you requested.

**Pull request merged**